test/spec/Feature/Query/RawSQLSpec.hs

```diff
@@ -0,0 +1,43 @@
+module Feature.Query.RawSQLSpec where
+
+import Network.Wai (Application)
+
+import Network.HTTP.Types
+import Network.Wai.Test    (SResponse (simpleHeaders, simpleStatus))
+import Test.Hspec
+import Test.Hspec.Wai
+import Test.Hspec.Wai.JSON
+
+import Protolude
+import SpecHelper (acceptHdrs)
+
+spec :: SpecWith ((), Application)
+spec = describe "When accept header is set to application/json+sql" $ do
+  let mtApplicationJSONSQLHdrs = acceptHdrs "application/json+sql"
+
+  it "responds raw sql to a GET request" $ do
+    r <- request methodGet "/items?id=eq.1" mtApplicationJSONSQLHdrs ""
+    liftIO $ do
+      simpleStatus r `shouldBe` status200
+      simpleHeaders r `shouldContain` [("Content-Type", "application/json; charset=utf-8")]
```

Comment:
We would need some samples of the response bodies in these tests.

For now I've checked it locally and it looks like:

```bash
curl "localhost:3000/projects?id=eq.1&name=eq.Joe" -H "Accept: application/json+sql"

{"params":["1","Joe"],"sql":"WITH pgrst_source AS ( SELECT \"test\".\"projects\".* FROM \"test\".\"projects\" WHERE  \"test\".\"projects\".\"id\" = $1 AND  \"test\".\"projects\".\"name\" =
 $2  )  SELECT null::bigint AS total_result_set, pg_catalog.count(_postgrest_t) AS page_total, ''::text AS body, nullif(current_setting('response.headers', true), '') AS response_headers, 
nullif(current_setting('response.status', true), '') AS response_status, '' AS response_inserted FROM ( SELECT * FROM pgrst_source ) _postgrest_t"}
```

It's an interesting functionality for sure.

Reply:
Will got it added!